src/librustc_mir/borrow_check/mod.rs

```diff
@@ -1698,18 +1713,26 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
     fn move_path_closest_to<'a>(
         &mut self,
         place: &'a Place<'tcx>,
-    ) -> Result<(&'a Place<'tcx>, MovePathIndex), NoMovePathFound> where 'cx: 'a {
-        let mut last_prefix = place;
-        for prefix in self.prefixes(place, PrefixSet::All) {
-            if let Some(mpi) = self.move_path_for_place(prefix) {
-                return Ok((prefix, mpi));
+    ) -> Result<(Place<'tcx>, MovePathIndex), NoMovePathFound> where 'cx: 'a {
+        let mut last_prefix = place.base.clone();
+
+        for prefix in self.prefixes(&place.base, &place.projection, PrefixSet::All) {
+            if let Some(mpi) = self.move_path_for_place(&Place {
```

Comment:
can you make `move_path_for_place` clone-free, by giving it two args?

Reply:
The idea is to do that in this same PR, next commits. Unsure if you still want me to do that now.

Reply:
nope that's fine to do later